Wikipedia/Networking/Fetchers/MWKImageInfoFetcher.m

```diff
@@ -46,49 +47,88 @@ - (instancetype)initWithDelegate:(id<FetchFinishedDelegate>)delegate
     return self;
 }
 
-- (id<MWKImageInfoRequest>)fetchInfoForPageTitles:(NSArray*)imageTitles
+- (id<MWKImageInfoRequest>)fetchInfoForImagesFoundOnPages:(NSArray*)pageTitles
+                                                 fromSite:(MWKSite*)site
+                                         metadataLanguage:(NSString*)metadataLanguage
+                                           thumbnailWidth:(NSUInteger)thumbnailWidth
+                                                  success:(void (^)(NSArray*))success
+                                                  failure:(void (^)(NSError*))failure {
+    return [self fetchInfoForTitles:pageTitles
+                           fromSite:site
+                     thumbnailWidth:thumbnailWidth
+                   metadataLanguage:metadataLanguage
+                       useGenerator:YES
+                            success:success
+                            failure:failure];
+}
+
+- (id<MWKImageInfoRequest>)fetchInfoForImageFiles:(NSArray*)imageTitles
                                          fromSite:(MWKSite*)site
                                           success:(void (^)(NSArray*))success
                                           failure:(void (^)(NSError*))failure {
-    NSParameterAssert([imageTitles count]);
-    NSAssert([imageTitles count] <= 50, @"Only 50 titles can be queried at a time.");
+    // This is currently only used for fullscreen image gallery.  If more dynamic image sizes are desired, expose
+    // the parameter in the API.
+    // 1280 is a well-populated image width in back-end cache that gives good-enough quality on most iOS devices
+    return [self fetchInfoForTitles:imageTitles
+                           fromSite:site
+                     thumbnailWidth:1280
```

Comment:
to be clear: not used for the feed (see comment). also, I believe you were the one who recommended I use 1280 originally, as I used to fetch the original file (which could be much, _much_ larger!)